stellargraph/mapper/sliding.py

```diff
@@ -0,0 +1,138 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = [
+    "SlidingFeaturesNodeGenerator",
+    "SlidingFeaturesNodeSequence",
+]
+
+import numpy as np
+from . import Generator
+from tensorflow.keras.utils import Sequence
+
+from ..core.validation import require_integer_in_range
+
+
+class SlidingFeaturesNodeGenerator(Generator):
+    def __init__(self, G, window_size, batch_size=1):
+        require_integer_in_range(window_size, "window_size", min_val=1)
+        require_integer_in_range(batch_size, "batch_size", min_val=1)
+
+        self.graph = G
+
+        node_type = G.unique_node_type(
+            "G: expected a graph with a single node type, found a graph with node types: %(found)s"
+        )
+        self._features = G.node_features(node_type=node_type)
+
+        self.window_size = window_size
+        self._batch_size = batch_size
+
+    def num_batch_dims(self):
+        return 1
+
+    def flow(self, sequence_iloc_slice, target_distance=None):
+        return SlidingFeaturesNodeSequence(
+            self._features,
+            self.window_size,
+            self._batch_size,
+            sequence_iloc_slice,
+            target_distance,
+        )
+
+
+class SlidingFeaturesNodeSequence(Sequence):
+    def __init__(
+        self, features, window_size, batch_size, sequence_iloc_slice, target_distance
+    ):
+        if target_distance is not None:
+            require_integer_in_range(target_distance, "target_distance", min_val=1)
+
+        if not isinstance(sequence_iloc_slice, slice):
+            raise TypeError(
+                f"sequence_iloc_slice: expected a slice(...) object, found {type(sequence_iloc_slice).__name__}"
+            )
+
+        if sequence_iloc_slice.step not in (None, 1):
+            raise TypeError(
+                f"sequence_iloc_slice: expected a slice object with a step = 1, found step = {sequence_iloc_slice.step}"
+            )
+
+        self._features = features[:, sequence_iloc_slice, ...]
+        shape = self._features.shape
+        self._num_nodes = shape[0]
+        self._num_sequence_samples = shape[1]
+        self._num_sequence_variates = shape[2:]
+
+        self._window_size = window_size
+        self._target_distance = target_distance
+        self._batch_size = batch_size
+
+        query_length = window_size + (0 if target_distance is None else target_distance)
+        self._num_windows = self._num_sequence_samples - query_length + 1
+
+        # if there's not enough data to fill one window, there's a problem!
+        if self._num_windows <= 0:
+            if target_distance is None:
+                target_str = ""
+            else:
+                target_str = f" + target_distance={target_distance}"
+
+            total_sequence_samples = features.shape[1]
+            start, stop, step = sequence_iloc_slice.indices(total_sequence_samples)
+            assert step == 1
```

Comment:
What's the purpose of this assert? Hasn't it already been checked that `sequence_iloc_slice` has `step=1` on line 69, or is it more for readability?

Reply:
Yeah, for readability: 

- it's an assert (rather than a real `throw ValueError(...)`) because it is just syncing with the code on line 69 (e.g. if that one is changed, this one should be too)
- it's communicating that `step` is being ignored on purpose, because it has an uninteresting value

I'll add a brief comment.